components/admin_console/team_channel_settings/group/group_users/index.ts

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+import {bindActionCreators, Dispatch, ActionCreatorsMapObject} from 'redux';
+import {createSelector} from 'reselect';
+
+import {UserProfile} from 'mattermost-redux/types/users';
+import {GenericAction, ActionFunc} from 'mattermost-redux/types/actions';
+
+import {filterProfilesMatchingTerm, profileListToMap} from 'mattermost-redux/utils/user_utils';
+
+import {filterProfiles} from 'mattermost-redux/selectors/entities/users';
+import {getMembersInTeams} from 'mattermost-redux/selectors/entities/teams';
+import {getChannelMembersInChannels} from 'mattermost-redux/selectors/entities/channels';
+import {getConfig} from 'mattermost-redux/selectors/entities/general';
+
+import {GlobalState} from 'types/store';
+import {loadChannelMembersForProfilesList, loadTeamMembersForProfilesList} from 'actions/user_actions.jsx';
+import {setModalSearchTerm, setModalFilters} from 'actions/views/search';
+
+import GroupUsers, {Filters, Memberships} from './group_users';
+
+type Props = {
+    members: UserProfile[];
+    scope: 'team' | 'channel';
+    scopeId: string;
+    total: number;
+};
+
+type Actions = {
+    loadTeamMembersForProfilesList: (profiles: UserProfile[], id: string, reloadAllMembers?: boolean) => Promise<{
+        data: boolean;
+    }>;
+    loadChannelMembersForProfilesList: (profiles: UserProfile[], id: string, reloadAllMembers?: boolean) => Promise<{
+        data: boolean;
+    }>;
+    setModalSearchTerm: (term: string) => Promise<{
+        data: boolean;
+    }>;
+    setModalFilters: (filters: Filters) => Promise<{
+        data: boolean;
+    }>;
+};
+
+const searchUsers = createSelector(
```

Comment:
This is kind of misusing `createSelector`. Instead, you can use `memoizeResult` from 'mattermost-redux/utils/helpers` to do the same thing, but in a more concise way.

Also, while you might not have multiple instances of `GroupUsers` mounted at once, it's good practice to create any memoized functions in a `makeMapStateToProps` so that multiple instances of the component don't overwrite each others' memoized results.

```typescript
import {memoizeResult} from 'mattermost-redux/utils/helpers';

function makeMapStateToProps() {
    const searchUsers = memoizeResult(((users: UserProfile[], term: string, filters: Filters, memberships: Memberships): UserProfile[] => {
        // ...
    });

    return function mapStateToProps(state: GlobalState, props: Props) {
        // ...
    };
}

export default connect(makeMapStateToProps, mapDispatchToProps)(GroupUsers);
```